### PR TITLE
US915147: Add support for liveness + readiness checks

### PR DESCRIPTION
**Pull request by rorytorneymf**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=915147

- The `WorkerFactory` interface contains a new `livenessCheck` method, which has a default implementation that returns 
`HealthResult.RESULT_HEALTHY`. A worker may optionally override this method to provide their own implementation of liveness.
- A new `/health-check?name=all&type=ALIVE` endpoint has been added on the default REST port (8080) to check if a worker is alive
- A new `/health-check?name=all&type=READY` endpoint has been added on the default REST port (8080) to check if a worker is ready
- See the [documentation](https://github.com/WorkerFramework/worker-framework/tree/develop/worker-core#health-checks-within-the-worker-framework) for more details.

### CODE REVIEW

File: release-notes-9.0.0.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+!not-ready-for-release!
```
Comment (dermot-hardy) [inline release-notes-9.0.0.md]: The 8.1.0 release notes file is still present.

File: worker-store-http/src/main/java/com/hpe/caf/worker/datastore/http/HttpDataStore.java [hunk 2]
```diff
@@ -74,7 +74,13 @@ public void shutdown()
     }
 
     @Override
-    public HealthResult healthCheck()
+    public HealthResult checkAlive()
+    {
+        return new HealthResult(HealthStatus.HEALTHY);
+    }
```
Comment (dermot-hardy) [inline worker-store-http/src/main/java/com/hpe/caf/worker/datastore/http/HttpDataStore.java]: As discussed we've decided to make this a default implementation at the `caf-common` level so this shouldn't be needed here.

**Review by dermot-hardy — CHANGES_REQUESTED** [review]

File: worker-core/readme.md [hunk 3]
```diff
@@ -580,7 +683,6 @@ the current input queue. Default is True.
       return resultQueue;
     }
 
-
```
Comment (dermot-hardy) [inline worker-core/readme.md]: Don't drop this blank line.  There are two blank lines between all the other methods.

File: worker-test/pom.xml [hunk 4]
```diff
@@ -174,7 +174,7 @@
                             <alias>keystore</alias>
                             <name>${project.artifactId}-keystore:${project.version}</name>
                             <build>
-                                <from>${projectDockerRegistry}/cafapi/opensuse-jre11</from>
+                                <from>${dockerHubPublic}/cafapi/opensuse-jre11:3.0.0</from>
```
Comment (dermot-hardy) [inline worker-test/pom.xml]: This doesn't seem right.  Is it a merge issue?

File: worker-test/pom.xml [hunk 5]
```diff
@@ -194,7 +194,7 @@
                         </image>
                         <image>
                             <alias>webdav</alias>
-                            <name>${projectDockerRegistry}/cloudesire/webdav</name>
+                            <name>${dockerHubPublic}/cloudesire/webdav</name>
```
Comment (dermot-hardy) [inline worker-test/pom.xml]: This doesn't seem right.  Is it a merge issue?

File: worker-test/pom.xml [hunk 6]
```diff
@@ -254,14 +254,14 @@
                             <alias>worker-test</alias>
                             <name>${targetDockerRegistryPath}/worker-test:${project.version}</name>
                             <build>
-                                <from>${projectDockerRegistry}/cafapi/opensuse-jre17</from>
+                                <from>${dockerHubPublic}/cafapi/opensuse-jre17:1</from>
```
Comment (dermot-hardy) [inline worker-test/pom.xml]: This doesn't seem right.  Is it a merge issue?

File: worker-test/pom.xml [hunk 7]
```diff
@@ -340,6 +340,7 @@
                                 <ports>
                                     <port>${worker.testadminport2}:8081</port>
                                     <port>${worker.testdebugport2}:5005</port>
+                                    <port>8999:8080</port>
```
Comment (dermot-hardy) [inline worker-test/pom.xml]: We shouldn't use fixed ports in the build as they will lead to conflicts on the build machine is the port is already in use.

File: worker-configs/src/main/java/com/hpe/caf/configs/HealthConfiguration.java [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+    /**
+     * The initial delay to use when first scheduling the liveness check.
+     */
+    @Min(0)
+    private int livenessInitialDelaySeconds = 15;
+
+    /**
+     * The interval on which to perform a liveness check for while in a healthy state.
+     */
+    @Min(60)
```
Comment (dermot-hardy) [inline worker-configs/src/main/java/com/hpe/caf/configs/HealthConfiguration.java]: Doesn't feel right for the minimum to be 60 here.  Same for a number of other configs also.

File: worker-core/src/main/java/com/hpe/caf/worker/core/WorkerApplication.java [hunk 9]
```diff
@@ -107,8 +118,10 @@ public void run(final WorkerConfiguration workerConfiguration, final Environment
         WorkerThreadPool wtp = WorkerThreadPool.create(workerFactory);
         final int nThreads = workerFactory.getWorkerThreads();
         ManagedWorkerQueue workerQueue = queueProvider.getWorkerQueue(config, nThreads);
+
```
Comment (dermot-hardy) [inline worker-core/src/main/java/com/hpe/caf/worker/core/WorkerApplication.java]:
```suggestion
```

**Review by dermot-hardy — CHANGES_REQUESTED** [review]

File: worker-configs/src/main/java/com/hpe/caf/configs/HealthConfiguration.java (hunk 8 above)
Comment (rorytorneymf) [reply worker-configs/src/main/java/com/hpe/caf/configs/HealthConfiguration.java]: I didn't want to allow users to be able to set this to a very low value, as that would result in the checks getting run continuously, which I though could be a performance problem. I choose 60 as I think that is the lowest we currently use for `healthChecks.intervalSeconds` in Verity/deploy. What would be a better minimum value here, 10 seconds?

Comment (dermot-hardy) [reply worker-configs/src/main/java/com/hpe/caf/configs/HealthConfiguration.java]: I'd take it all the way down to 1 or 2.  It is just a minimum &mdash; it's not like you're advising that configuration.

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**